presto-main/src/main/java/com/facebook/presto/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -207,11 +209,61 @@ private static int jdbcDataType(Type type)
         return Types.JAVA_OBJECT;
     }
 
-    private static Integer decimalDigits(Type type)
+    private static Integer columnSize(Type type)
     {
         if (type.equals(BIGINT)) {
-            return 19;
+            return 19;  // 2**63-1
+        }
+        if (type.equals(INTEGER)) {
+            return 10;  // 2**31-1
+        }
+        if (type.equals(SMALLINT)) {
+            return 5;   // 2**15-1
+        }
+        if (type.equals(TINYINT)) {
+            return 3;   // 2**7-1
+        }
+        if (type instanceof DecimalType) {
+            return ((DecimalType) type).getPrecision();
+        }
+        if (type instanceof VarcharType) {
+            return ((VarcharType) type).getLength();
+        }
+        if (type.equals(VARBINARY)) {
+            return Integer.MAX_VALUE;
+        }
+        if (type.equals(TIME)) {
+            return 8; // 00:00:00
+        }
+        if (type.equals(TIME_WITH_TIME_ZONE)) {
+            return 8 + 6; // 00:00:00+00:00
+        }
+        if (type.equals(DATE)) {
+            return 15; // 292278993-01-01 (Long.MAX_VALUE as millis)
+        }
+        if (type.equals(TIMESTAMP)) {
+            return 15 + 8;
+        }
+        if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
+            return 15 + 8 + 6;
+        }
+        return null;
+    }
+
+    // DICIMAL_DIGITS represents the number of fractional digits.
+    private static Integer decimalDigits(Type type)
```

Comment:
@electrum JavaDoc of `getColumns()` states that COLUMN_SIZE and DECIMAL_DIGITS returns null for data types where it is not applicable.

https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getColumns-java.lang.String-java.lang.String-java.lang.String-java.lang.String-

And for CHAR_OCTET_LENGTH I thought it's undefined because JavaDoc says nothing about null unlike other fields. But when I checked this time again I found that actually JavaDoc for similar method `getProcedureColumns()` states that it returns null as well. (JavaDoc bug?)

https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getProcedureColumns-java.lang.String-java.lang.String-java.lang.String-java.lang.String-

So I updated charOctetLength() as well to return null instead of 0 at 6bd94fd6d435885d070bf739f1869cd5f0b767db

I think we should keep these methods to have Integer as return value.